enhancements/installer/aws-load-balancer-type.md

```diff
@@ -0,0 +1,327 @@
+
+### Goals
+
+As an administrator, I would like to specify the load balancer type in AWS either to NLB or CLB/ELB during installation.
+As an user/administrator, when an ingress controller CR is deleted then I would like the ingress controller to be recreated with the load balancer mentioned during installation.
+In short I would like the persistence of the load balancer type in the default ingress controller.
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+We will be adding a field awsLBType in the install-config for the users to provide a load balancer either to NLB or CLB/ELB
+during installation.
+
+### Workflow Description
+
+The installer will accept a LB type from user and then we would set type of LB in the following object.
+
+`oc get ingress.config.openshift.io/cluster -o yaml`
+
+Then we would fetch info from this object and reconcile the operator to have the NLB changes reflected.
+
+### User Stories
+
+#### Story 1
+
+As an admin I want to install a specific load balancer rather than the default CLB/ELB.
+
+#### Story 2
+
+As an admin I want the load balancer type to persist even if the default ingress controller object is deleted which was mentioned during the installation.
+
+### API Extensions
+
+### Implementation Details/Notes/Constraints
+
+For the install config object we will be adding a field called lbType to the [API](https://github.com/openshift/installer/blob/master/pkg/types/aws/platform.go#L7)
+
+It will look as follows -
+```go
+package aws
+
+import "github.com/aws/aws-sdk-go/aws/endpoints"
+
+// Platform stores all the global configuration that all machinesets
+// use.
+type Platform struct {
+
+// LBType allows user to set a load balancer type.
+// Allowed values are nlb or elb
+// +optional
+LBType string `json:"lbType,omitempty"`
+}
+```
+
+In the infra config cluster object we will need to add api for AwsLBType which will store and persist the information
+of the desired LB type mentioned by the installer config API so that the ingress operator will refer the LB type from this object whenever someone accidentally deletes the ingress controller.
+Currently, the default LB type i.e CLB/ELB gets set if you delete the ingress controller object even when the desired the LB type for default ingress controller
+was set to NLB as ingress operator does not have a way to check the LB type as we don't store it.
+
+The API will look as follows -
+Code will be [here](https://github.com/openshift/api/blob/63b58097a95d1b053cf038f56f89afc42524f598/config/v1/types_infrastructure.go#L351)
+```go
+// AWSPlatformSpec holds the desired state of the Amazon Web Services infrastructure provider.
+// This only includes fields that can be modified in the cluster.
+type AWSPlatformSpec struct {
+// LBType is used to set type of the LB either to CLB/ELB or NLB for the 
+// default ingress controller in AWS environment.
+// +optional
+  LBType string `json:"lbType,omitempty"`
```

Comment:
Are you likely to ever want to expand load balancer configuration to handle other options for the NLBs or ELBs? We may want to make this a discriminated union instead

Reply:
no

Reply:
How can you be certain about that? What are the other relevant Load balancer options that are available from AWS, does the ingress object expose anything?

Reply:
There are timeout options available for classic but we don't set any options for NLB. However we can have something like 

https://github.com/openshift/api/pull/1209#discussion_r936724881

which you mentioned.

ingress:
  loadBalancer:
    defaultType: NLB

Reply:
BTW there are options like loadBalancerSourceRanges and Scope for AWS LB types.